exporter/datadogexporter/traces_exporter_test.go

```diff
@@ -40,24 +38,82 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/testutils"
 )
 
-func testTracesExporterHelper(td ptrace.Traces, t *testing.T) []string {
+func TestMain(m *testing.M) {
+	tracelog.SetLogger(&testlogger{})
+	os.Exit(m.Run())
+}
+
+type testlogger struct{}
```

Comment:
That idiom is used when there is no code available which would fail compilation if the interface would not be implemented, for example: when implementing an interface in one package and using it in another. In out case, the interface is used in the same package and compilation would fail if it would not be implemented correctly, so I don't think it's necessary to add that check. If your intention is simply to indicate to the reader what the implemented interface is, then perhaps documentation is a better, less clever and less cryptic way to achieve that.

Reply:
I am just pointing to a de facto convention on this repository: `rg 'var _' ` returns 282 instances of the pattern in this repo. I think it is still useful to have this even if used elsewhere, since this 'assertion' will be kept no matter what refactoring you do of other parts of the code. In any case, I don't want to block the PR because of such a minor thing so if you don't want to do it that's fine too.

Reply:
No problem. If it's a convention of the repository, I will of course do it!